### ISSUE TITLE
Unable to compile kitty on macOS

### ISSUE THREAD

**Issue by Luflosi** [kovidgoyal/kitty#1196]
The compilation fails with
```
In file included from /path/to/kitty/kitty/core_text.m:14:
In file included from /usr/local/Cellar/harfbuzz/2.2.0/include/harfbuzz/hb-ot.h:33:
In file included from /usr/local/Cellar/harfbuzz/2.2.0/include/harfbuzz/hb-ot-color.h:37:
/usr/local/Cellar/harfbuzz/2.2.0/include/harfbuzz/hb-ot-name.h:79:22: error: 
      redefinition of typedef 'hb_ot_name_id_t' is a C11 feature
      [-Werror,-Wtypedef-redefinition]
typedef unsigned int hb_ot_name_id_t;
                     ^
/usr/local/Cellar/harfbuzz/2.2.0/include/harfbuzz/hb-deprecated.h:244:22: note: 
      previous definition is here
typedef unsigned int hb_ot_name_id_t; /* Since is in hb-ot.h */
                     ^
1 error generated.
make: *** [app] Error 1
```
This might be an issue with harfbuzz.

**Comment by kovidgoyal** [comment]
that is an issue with harfbuzz, not kitty. Stick to older harfbuzz releases until they fix it.

**Comment by Luflosi** [comment]
I case anyone else has this problem,
```Bash
brew switch harfbuzz 2.1.3
```
worked for me.

**Comment by Luflosi** [comment]
@kovidgoyal should I file a bug report for harfbuzz or do they know about this?

**Comment by kovidgoyal** [comment]
@Luflosi no idea, I was planning to do it myself, but am totally swamped at the moment, so I just disabled the kitty travis build with brew for the moment.

**Comment by Luflosi** [comment]
I filed the bug report.

**Comment by jacwah** [comment]
Another solution is to compile in c11 mode: `CFLAGS="-std=c11" make`.

**Comment by Luflosi** [comment]
It's fixed in harfbuzz 2.3.0, which is now available via homebrew.
You may (or may not, I don't really care) revert your commit if you like.

**Comment by kovidgoyal** [comment]
I'm ok with keeping it, unless someone is harmed by it.
